content/spire/try/oidc-federation-aws.md

```diff
@@ -0,0 +1,448 @@
+---
+title: AWS OIDC Authentication
+description: Using SPIRE and OIDC to Authenticate Workloads on Kubernetes to AWS S3
+weight: 4
+toc: true
+aliases:
+menu:
+  spire:
+    weight: 50
+    parent: 'spire-try'
+---
+
+This tutorial builds on the [Kubernetes Quickstart](/spire/try/getting-started-k8s/) guide to describe how a SPIRE identified workload can, using a JWT-SVID, authenticate to Amazon AWS APIs, assume an AWS IAM role, and retrieve data from an AWS S3 bucket. This avoids the need to create and deploy AWS IAM credentials with the workload itself.
+
+In this tutorial you will learn how to:
+
+* Deploy the OIDC Discovery Provider Service
+* Create the required DNS A record to point to the OIDC Discovery document endpoint
+* Create a sample AWS identity provider, policy, role, and S3 bucket
+* Test access to the S3 bucket
+
+## Prerequisites
+
+Note the following required accounts, prerequisites, and limitations before starting this tutorial:
+
+* You'll need access to the Kubernetes environment that you configured when going through [Kubernetes Quickstart](/spire/try/getting-started-k8s/). The Kubernetes environment must be able to expose an Ingress to the public internet. _Note: This is generally not true for local Kubernetes environments such as Minikube._
+* You'll need access to the AWS console with an account that has permissions to configure an identity provider, IAM policy, IAM role, and S3 bucket.
+* You'll need the ability to configure a DNS A record for the SPIRE OIDC Discovery document endpoint (see [Part 2](#part-2-configure-dns-for-the-oidc-discovery-provider-ip-address))
+
+
+# Part 1: Configure SPIRE Components
+
+In the first part of this procedure, you configure the SPIRE components in Kubernetes.
+
+## Download Kubernetes YAML Files for this Tutorial
+
+To get the YAML files required for this tutorial, clone https://github.com/spiffe/spire-tutorials. The files for this tutorial are in the **k8s/oidc-aws** directory.
```

Comment:
is there any rule of thumb to use `code` formatting? I'm used to read documentation that uses it when referencing commands, urls and file paths

Reply:
Yeah, I agree we should use a monospace font for commands, URLs, file names, and paths. However, I'd prefer to leave live (clickable) URLs in the default bold+colored rendering automatically applied by Markdown/Hugo (e.g. https://jwt.io). The k8s names I've left in bold since the Quickstart for Kubernetes doc uses that convention, but eventually I think I'd like to use monospace for k8s names, too.

I've updated the doc and will push a commit soon.